lib/provisioning/Provisioner.js

```diff
@@ -129,24 +143,170 @@ Provisioner.prototype._validate = function(actual, parameterName) {
 };
 
 // Validate parameters for use in linking/unlinking
-Provisioner.prototype._validateAll = function(parameters) {
-    let parameterNames = ['matrix_room_id', 'remote_room_channel', 'remote_room_server'];
+Provisioner.prototype._validateAll = function(parameters, parameterNames) {
```

Comment:
This is getting sufficiently complex that I'm unhappy for it to continue in its current form. Look into using the `matrix-appservice-bridge` component [ConfigValidator](http://matrix-org.github.io/matrix-appservice-bridge/1.2.0/ConfigValidator.html) and write a schema to represent the `/link` input JSON. You can then use it like so:
```javascript
var validator = new ConfigValidator({
  // .. schema as a JSON object
});
try {
  validator.validate(req.body);
}
catch (err) {
  // handle validation failures
}
```